### ISSUE TITLE
How to run this code

### ISSUE THREAD

**Issue by YichiHuang** [YerevaNN/Spoken-language-identification#6]
Hi,
I got a speech dataset, the length of each recording ranges from 2s~5s. How I run your code with my dataset? Can you give me some advice. Thanks.

**Comment by hrayrhar** [comment]
Hi,

First you should create the spectrograms of the recordings (you can use `create_spectrograms.py` for that), then make training and validation list files like [this](https://gist.github.com/Harhro94/aa11fe6b454c614cdedea882fd00f8d7) and finally, run the `theano/main.py`.
As the length of recordings in your dataset is not constant you should either set `batch_size=1` or do something to equalize the lengths in mini-batch.

**Comment by YichiHuang** [comment]
Hi,
I followed your experiment procedure and run with my dataset. My dataset contains 50 classes, 400 recordings with 5 seconds for each category.
First of all, I created spectrograms of each recording with size in 256x429. Then I split them in training set(1500 in total) and validation set(500 in total).
Last, I run the `main.py` with default parameters except:
`--network==tc_net_mod` 
I got the result:
`accuracy: 0.59 percent`
`accuracy: 2.08 percent`
`saving ... states/tc_net_mod.b32.bn.epoch499.test3.91200.state`
The accuracy is extremely low and I wonder there is something wrong with my experiment.
Can you give some comments?
Thanks.

**Comment by ouceduxzk** [comment]
I also get very low acc running the code with the topcoder dataset even following the strategy you mentioned in the blog : adadelta first and then sgd

**Comment by MurugeshMarvel** [comment]
![screenshot from 2017-02-26 09-06-00](https://cloud.githubusercontent.com/assets/9530814/23336877/defb8a54-fc02-11e6-98f0-da90b1b3ed3f.png)
getting error like this

**Comment by manbharae** [comment]
@YichiHuang  can you please share your code. I trying to learn too.

**Comment by nitalternate** [comment]
hi Harhro94 ,

 I'm a beginner in this language recognition field. Now i'm practicing your code. I downloaded a data-set from 'http://www.topcoder.com/contest/problem/SpokenLanguages2/testingdata.zip' which is in mp3 format. I converted it into csv format. But in your code, you are using wav files, should i convert mp3 files into .wav files?

In create_spectrograms file, there is a function 'plotstft' , inside that function, you are calling a wavfile and after that you are taking a particular path. I'm using anaconoda(windows), can you please tell me , which path you are calling there.

Thank you...

**Comment by nitalternate** [comment]
@Harhro94  I'm a beginner in this language recognition field. Now i'm practicing your code from ' https://github.com/YerevaNN/Spoken-language-identification ' .

 I downloaded a data-set from 'http://www.topcoder.com/contest/problem/SpokenLanguages2/testingdata.zip' which is in mp3 format. I converted it into csv format. I'm using anaconda(windows), While executing create_spectrograms  file , spectrograms are not generating, and also the main function is showing "network_batch error"




Are you using create_spectrograms  & main files for language recognition? Only those two files?

Please let me know the answers of  my questions.



Expecting reply...  

Thank you...
